platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/DefaultUsersDeletionScheduler.java

```diff
@@ -0,0 +1,235 @@
+      return false;
+    }
+    if (context != null && context.getRouteStatus(ROUTE_ID) != ServiceStatus.Started) {
+      try {
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .routeId(ROUTE_ID)
+                    .bean(DefaultUsersDeletionScheduler.class, "removeDefaultUsers");
+              }
+            };
+        context.addRoutes(builder);
+        context.start();
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create camel route.", e);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @VisibleForTesting
+  String getCron() {
+    if (!getTempTimestampFilePath().toFile().exists()) {
+
+      // Create temp file and add timestamp
+      Instant instant = Instant.now();
+      if (!createTempFile(instant)) {
+        return null;
+      }
+      return cronCalculator(instant);
+
+    } else {
+
+      // Read timestamp
+      Instant firstInstall = installationDate();
+      if (firstInstall == null) {
+        return null;
+      }
+
+      if (Instant.now()
+          .isAfter(
+              firstInstall
+                  .plus(Duration.ofDays(3))
+                  .minus(Duration.ofMinutes(30)))) { // Timestamp expired
+        removeDefaultUsers();
+        return null;
+      }
+
+      return cronCalculator(firstInstall);
+    }
+  }
+
+  private boolean createTempFile(Instant instant) {
+    try {
+      if (getTempTimestampFilePath().toFile().createNewFile()) {
+        return writeTimestamp(instant);
+      }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to access the temporary file", e);
+    }
+    return false;
+  }
+
+  private boolean writeTimestamp(Instant instant) throws IOException {
+    try (ObjectOutputStream objectOutputStream =
+        new ObjectOutputStream(new FileOutputStream(getTempTimestampFilePath().toFile()))) {
+      objectOutputStream.writeObject(instant);
+      return true;
+    }
+  }
+
+  public static boolean removeDefaultUsers() {
+    try {
+      Bundle bundle = FrameworkUtil.getBundle(DefaultUsersDeletionScheduler.class);
+      if (bundle != null) {
+        BundleContext bundleContext = bundle.getBundleContext();
+        Collection<ServiceReference<BackingEngineFactory>> implementers =
+            bundleContext.getServiceReferences(BackingEngineFactory.class, null);
+        for (ServiceReference<BackingEngineFactory> impl : implementers) {
+
+          BackingEngineFactory backingEngineFactory = bundleContext.getService(impl);
+          BackingEngine backingEngine =
+              backingEngineFactory.build(
+                  ImmutableMap.of("users", getUsersPropertiesFilePath().toString()));
+
+          if (!backingEngine.listUsers().isEmpty()) {
+            backingEngine.listUsers().forEach(user -> backingEngine.deleteUser(user.getName()));
```

Comment:
Did we test this against LDAP to make sure that we don't try to delete LDAP users with the same names as those found in users.properties?

Reply:
I manually verified that LDAP isn't listed as an implementation of the `BackingEnginFactory` so it won't delete the LDAP users.